### PR TITLE
Loyalty program

### PR DESCRIPTION
**Pull request by dvinubius**
Opened for discussion

### CODE REVIEW

File: contracts/LoyaltyCardMaster.sol [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal override {
+        // this only goes through if token is being sent to a WL destination
+        // or a WL destination (related to Impossible Finance) transfers it (back)
+        if (!whitelistedDestination[to] && !whitelistedDestination[from])
+            revert NotAllowedAsDestination();
+        super._transfer(from, to, tokenId);
+    }
```
Comment (friday-james) [inline contracts/LoyaltyCardMaster.sol:206]: We might need to check if the receiver owns a loyalty card. Or is it done somewhere else?

Comment (dvinubius) [reply contracts/LoyaltyCardMaster.sol:206]: Receiver must be a whitelistedDestination. We'll only whitelist accounts representing IF staking contracts. 
If it weren't for this feature of staking loyalty cards, we wouldn't allow any transfers at all.
So the receiver will typically be a contract and typically "own" multiple loyalty cards, even though their original owners remain the IF users associated with them

**Comment by dvinubius** [comment]
@acceleratedmonkey If we merge this code then I can proceed with the rewarder contracts (which use the NFT lookup functionality being reviewed right here)

File: contracts/LoyaltyCardMaster.sol (hunk 1 above)
Comment (friday-james) [reply contracts/LoyaltyCardMaster.sol:206]: I see 😮 😮

**Pull request closed without merging**